iothub/device/src/Transport/Amqp/AmqpTransportHandler.cs

```diff
@@ -548,17 +548,12 @@ protected override void Dispose(bool disposing)
             {
                 base.Dispose(disposing);
 
-                if (_disposed)
```

Comment:
Why is this a problem?

Reply:
+1. We seem to set _disposed to true only after removing the amqp unit, so it shouldn't cause leaks, right?

Reply:
@abhipsaMisra - Are you not OOF :D Take rest!

Reply:
_disposed is a protected field from the base class. That's what's throwing us all off

Reply:
when we call base.Dispose(), the _disposed is getting set to false, then we bail out prematurely.

Reply:
I moved the dispose to below the if statement to match the pattern of the other handlers.